pkgs/development/python-modules/lightgbm/default.nix

```
@@ -28,16 +34,25 @@ buildPythonPackage rec {
 
   dontUseCmakeConfigure = true;
 
-  buildInputs = lib.optionals stdenv.cc.isClang [
+  buildInputs = (optionals stdenv.cc.isClang [
     llvmPackages.openmp
-  ];
+  ]) ++ (optionals gpuSupport [
+    boost
+    ocl-icd
+    opencl-headers
+  ]);
 
   propagatedBuildInputs = [
     numpy
     scipy
     scikit-learn
   ];
 
+  buildPhase = concatStringsSep " " ([
+    python.pythonForBuild.interpreter
+    "setup.py bdist_wheel"
+  ] ++ optionals gpuSupport ["--gpu"]);
```

```suggestion
  buildPhase = ''
    ${python.pythonForBuild.interpreter} setup.py bdist_wheel ${lib.optionalString gpuSupport "--gpu"
  '';
```
How about we convert this into a string? That looks way less scary and is already done twice in tree.